core/schema/src/main/java/org/hawkular/metrics/schema/SchemaService.java

```diff
@@ -39,67 +41,125 @@
 
     private static Logger logger = LoggerFactory.getLogger(SchemaService.class);
 
-    public void run(Session session, String keyspace, boolean resetDB) {
-        run(session, keyspace, resetDB, 1);
+    private PreparedStatement versionUpdateQuery;
+
+    private Session session;
+
+    private String keyspace;
+
+    private Cassalog cassalog;
+
+    public SchemaService(Session session, String keyspace) {
+        this.keyspace = keyspace;
+        this.session = session;
     }
 
-    public void run(Session session, String keyspace, boolean resetDB, int replicationFactor) {
-        run(session, keyspace, resetDB, replicationFactor, false);
+    public void run(boolean resetDB) {
+        run(resetDB, 1);
     }
 
-    public void run(Session session, String keyspace, boolean resetDB, int replicationFactor, boolean updateVersion) {
+    public void run(boolean resetDB, int replicationFactor) {
+        run(resetDB, replicationFactor, false);
+    }
+
+    public void run(boolean resetDB, int replicationFactor, boolean updateVersion) {
```

Comment:
SchemaService has evolved a bit since it was originally written. If you want feel free to refactor this and inject ```resetDB```, ```replicationFactor```, and ```updateVersion``` in the constructor and/or setter methods. Then we won't have to call ```buildCassalogIfNeeded``` in multiple places.

Reply:
They keyspace does not have to exist in advance. Building Cassalog only initializes its properties. Cassalog does not try connecting to Cassandra on initialization.